contracts/p1/BasketHandler.sol

```diff
@@ -534,8 +541,45 @@ contract BasketHandlerP1 is ComponentP1, IBasketHandler {
         emit BasketSet(nonce, basket.erc20s, refAmts, disabled);
     }
 
+    /// Require that newERC20s and newTargetAmts preserve the current config targets
+    function requireConstantConfigTargets(
+        IERC20[] calldata newERC20s,
+        uint192[] calldata newTargetAmts
+    ) private {
+        // Empty _targetAmts mapping
+        while (_targetAmts.length() > 0) {
+            (bytes32 key, ) = _targetAmts.at(0);
+            _targetAmts.remove(key);
+        }
+
+        // Populate _targetAmts mapping with old basket config
+        uint256 len = config.erc20s.length;
+        for (uint256 i = 0; i < len; ++i) {
+            IERC20 erc20 = config.erc20s[i];
+            bytes32 targetName = config.targetNames[erc20];
+            (bool contains, uint256 amt) = _targetAmts.tryGet(targetName);
+            _targetAmts.set(
+                targetName,
+                contains ? amt + config.targetAmts[erc20] : config.targetAmts[erc20]
+            );
+        }
+
+        // Require new basket is exactly equal to old basket, in terms of targetAmts by targetName
+        len = newERC20s.length;
+        for (uint256 i = 0; i < len; ++i) {
+            bytes32 targetName = assetRegistry.toColl(newERC20s[i]).targetName();
+            // if the asset registry has a new registered asset targetName, this always reverts
```

Comment:
Its not clear if the comment is for the line on top or botton. Maybe remove the empty line? Also suggesting added a test case for this.

Reply:
I think I prefer having the comment below in this case. Having a clarifying comment first (as opposed to a descriptive or summary comment) makes the code less readable imo. I agree it's not totally clear what the comment applies to though, so I'll add some more detail to the text.

Reply:
I see though...this comment is super ambiguous in what it applies to. Removing and adding a test case.